x-pack/plugin/core/src/main/java/org/elasticsearch/license/XPackLicenseState.java

```diff
@@ -593,197 +542,160 @@ public synchronized boolean isGraphAllowed() {
      * @return {@code true} as long as the license is valid. Otherwise
      *         {@code false}.
      */
-    public synchronized boolean isMachineLearningAllowed() {
-        final Status currentStatus = status;
-        return currentStatus.active && isMachineLearningAllowedForOperationMode(currentStatus.mode);
+    public boolean isMachineLearningAllowed() {
+        return isAllowedByLicenseAndSecurity(OperationMode.PLATINUM, false, true, true);
     }
 
     public static boolean isMachineLearningAllowedForOperationMode(final OperationMode operationMode) {
-        return isPlatinumOrTrialOperationMode(operationMode);
+        return isAllowedByOperationMode(operationMode, OperationMode.PLATINUM, true);
     }
 
     /**
      * Transform is always available as long as there is a valid license
      *
      * @return true if the license is active
      */
-    public synchronized boolean isTransformAllowed() {
-        return status.active;
+    public boolean isTransformAllowed() {
+        return isActive();
     }
 
     public static boolean isTransformAllowedForOperationMode(final OperationMode operationMode) {
         // any license (basic and upwards)
         return operationMode != License.OperationMode.MISSING;
     }
 
+    public static boolean isFipsAllowedForOperationMode(final OperationMode operationMode) {
+        return isAllowedByOperationMode(operationMode, OperationMode.PLATINUM, true);
+    }
```

Comment:
It bothers me that we have this method, and them re-implement the same `OperationMode` check for FIPS in `LicenseService`